news/1258-default-cast-b8.rst

```diff
@@ -0,0 +1,42 @@
+New features
+------------
+
+* H5T_BITFIELD types will now be cast to their ``numpy.uint`` equivelant by default
+  (:issue:`1258`). This means that no knowledge of mixed type compound dataset
+  schemas is required to read these types, and can simply be read as follows:
+
+  .. code::
+
+     arr = dset[:]
+
+  Alternatively, these types can still be cast to ``numpy.bool`` explicitly:
+
+  .. code::
+
+     with dset.astype(numpy.bool):
+        arr = dset[:]
```

Comment:
```suggestion
  Alternatively, 8-bit bitfields can still be cast to booleans explicitly:

  .. code::

     arr = dset.astype(numpy.bool_)[:]:
```

`numpy.bool` recently started giving a deprecation warning. And I'd eventually like to shuffle people away from using `astype()` as a context manager now that it can be used as a wrapper.